### PR TITLE
Extend Documentation

### PR DESCRIPTION
**Pull request by swissspidy**
I tried to leverage Docusaurus to create a simple website containing documentation etc. for Traduttore.

Here's the current result: https://wearerequired.github.io/traduttore/

If we decide that it's too much of a hassle it's not a big deal. At least now we have some written documentation.

See #39.

### CODE REVIEW

File: docs/hooks.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+**Parameters:**
+
+* `$credentials`: Git credentials in the form `username:password`. Default empty string.
+* `$repository`: The current repository.
+
+----
+
+### `traduttore.zip_generated_send_notification`
+
+**Since:** 2.0.3
```
Comment (grappler) [inline docs/hooks.md]:
```suggestion
**Since:** 3.0.0
```

File: docs/hooks.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+* `$send_message`: Whether to send a notification or not. Default true.
+* `$translation_set`: Translation set the ZIP is for.
+* `$project`: The GlotPress project that was updated.
+
+----
+
+### `traduttore.zip_generated_notification_message`
+
+**Since:** 2.0.3
```
Comment (grappler) [inline docs/hooks.md]:
```suggestion
**Since:** 3.0.0
```

File: docs/hooks.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+* `$message`: The notification message.
+* `$translation_set`: Translation set the ZIP is for.
+* `$project`: The GlotPress project that was updated.
+
+----
+
+### `traduttore.updated_send_notification`
+
+**Since:** 2.0.3
```
Comment (grappler) [inline docs/hooks.md]:
```suggestion
**Since:** 3.0.0
```

File: docs/hooks.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+**Parameters:**
+
+* `$send_message`: Whether to send a notification or not. Defaults to true, unless there were no string changes at all.
+* `$project`: The GlotPress project that was updated.
+* `$stats`: Stats about the number of imported translations.
+
+----
+
+### `traduttore_updated_from_github_notification_message`
```
Comment (grappler) [inline docs/hooks.md]:
```suggestion
### `traduttore.updated_notification_message`
```

File: docs/hooks.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+**Since:** 2.0.3
```
Comment (grappler) [inline docs/hooks.md]:
```suggestion
**Since:** 3.0.0
```

File: docs/hooks.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+* `$translation_set`: Translation set the ZIP is for.
+* `$project`: The GlotPress project that was updated.
+
+----
+
+### `traduttore.updated_send_notification`
+
+**Since:** 2.0.3
+
+Filters whether a Slack notification for translation updates from GitHub should be sent.
```
Comment (grappler) [inline docs/hooks.md]: We should mention that the Slack notifications require [the plugin](https://wordpress.org/plugins/slack/) to work.

File: docs/installation.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+## System Requirements
+
+Traduttore is a WordPress plugin that sits on top of the [GlotPress](https://glotpress.org/) plugin. That means you need to have both WordPress and GlotPress installed.
+
+GlotPress is available as a WordPress plugin through the plugin directory. Installing it is as simple as searching for “GlotPress” and installing it. After activating the plugin, GlotPress can be accessed via `https://<home_url>/glotpress/`.
+
+Traduttore requires at least PHP 7.1, while the [Traduttore Registry](https://github.com/wearerequired/traduttore-registry) also supports PHP 7.0.
+
+Traduttore requires [WP-CLI](https://wp-cli.org/) 2.0 or newer to be installed on the server. [Learn more about the available CLI commands](cli.md).
+
```
Comment (grappler) [inline docs/installation.md]: We could add the Slack plugin requirement here.

File: docs/integration.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+After that, you can use `Required\Traduttore_Registry\add_project( $type, $slug, $api_url )` in your theme or plugin. On a multisite install, it's recommend to use it in a must-use plugin instead.
+
+**Note:** Alternatively, you could copy the library's code to your project.
+
+### Example
+
+Here's an example of how you can use `add_project()` in your plugin or theme:
+
+```php
+Required\Traduttore_Registry\add_project(
```
Comment (grappler) [inline docs/integration.md]: Missing the `\`  in the beginning.

**Comment by swissspidy** [comment]
> I am not really warm to Docusaurus.

What exactly bugs you? The layout, the way things are organized, generating the docs site, or ... ?

We don't have to use Docusaurus of course, but before I look into alternatives I want to know in which areas they need to be better.

**Comment by grappler** [comment]
> What exactly bugs you?
One is the 🦖and another is the homepage at the moment. And the styling of the links of not existing styling.
image

It is something new that I have not used before. It is easier to just use the docs folder in GitHub. I wonder if it would be better if it was integrated in our Website.

**Comment by swissspidy** [comment]
> One is the 🦖and another is the homepage at the moment. And the styling of the links of not existing styling.

This is just the default styling of a Docusauraus generated site... It's not like I added a dinosaur on purpose or something 😉 

I haven't spent any time on styling that site becuase it would be wasted time if we decide against using it.

Of course using GitHub Pages with a simple default theme and some Markdown files in the docs folder is way easier. If that's enough for us, perfect.

However, I don't think this really is something for our website. Some introduction to the plugin (like we already have for others) yes, but certainly not technical documentation.

File: docs/installation.md (hunk 7 above)
Comment (swissspidy) [reply docs/installation.md]: Adding a separate doc for that instead.

**Comment by swissspidy** [comment]
The default Jekyll themes available on GitHub are quite ugly.

doscify (https://docsify.js.org/#/themes?id=click-to-preview) is way cleaner and would allow for a more simple, straightforward documentation site. However, it's a SPA that requires JavaScript.

Alternatively we can just run with Docusaurus for now and switch to something else later.

@ocean90 @grappler Thoughts?

**Comment by grappler** [comment]
> The default Jekyll themes available on GitHub are quite ugly.

True but this one is not too bad. https://pages-themes.github.io/cayman/

> Alternatively we can just run with Docusaurus for now and switch to something else later.

My preference would be not to have anything and just merge the docs improvements and look at the website later.

**Comment by swissspidy** [comment]
> True but this one is not too bad. pages-themes.github.io/cayman

Saw that one too but everything is on one page and if you add multiple pages then you have to roll your own navigation etc.

**Pull request merged**